src/com/gfredericks/test/chuck/clojure_test.cljc

```diff
@@ -0,0 +1,104 @@
+(ns com.gfredericks.test.chuck.clojure-test
+  (:require [clojure.test.check :as tc]
+            [clojure.test.check.properties :as prop
+             #?@(:cljs [:include-macros true])]
+            #?(:clj  [clojure.test :refer [is]]
+               :cljs [cljs.test :refer-macros [is]])))
+
+;; copied from clojure.test.check, which privatized the function in
+;; recent versions.
+;;
+;; I think there might be plans for test.check to abstract this logic
+;; into a protocol or something, so I'm not too bothered by the
+;; copypasta for now.
+(defn ^:private not-exception?
+  [value]
+  (not (instance? #?(:clj  Throwable
+                     :cljs js/Error)
+                  value)))
+
+(defn report-exception [result]
+  (is (not-exception? (:result result)) result))
+
+(defn pass? [reports]
+  (every? #(= (:type %) :pass) reports))
+
+(defn report-needed? [reports final-reports]
+  (or (not (pass? reports)) (empty? final-reports)))
+
+(defn save-to-final-reports [final-reports reports]
+  (if (report-needed? reports final-reports)
+    reports
+    final-reports))
+
+(def ^:dynamic *chuck-captured-reports*)
+
+#?(:cljs
+(defmethod cljs.test/report [::chuck-capture :fail]
+  [m]
+  (swap! *chuck-captured-reports* conj m)))
+
+#?(:cljs
+(defmethod cljs.test/report [::chuck-capture :pass]
+  [m]
+  (swap! *chuck-captured-reports* conj m)))
+
+(defn capture-reports*
+  [reports-atom f]
+  #?(:clj
+     (binding [clojure.test/report #(swap! reports-atom conj %)]
+       (f))
+
+     :cljs
+     (binding [*chuck-captured-reports* reports-atom
+               cljs.test/*current-env* (cljs.test/empty-env ::chuck-capture)]
+       (f))))
+
+(defmacro capture-reports
+  [& body]
+  `(let [reports# (atom [])]
+     (capture-reports* reports# (fn [] ~@body))
+     @reports#))
+
+(defmacro qc-and-report-exception
+  [final-reports tests bindings & body]
+  `(report-exception
+    (tc/quick-check
+      ~tests
+      (prop/for-all ~bindings
+        (let [reports# (capture-reports ~@body)]
+          (swap! ~final-reports save-to-final-reports reports#)
+          (pass? reports#))))))
+
+(defn -testing
+  [name func]
+  (#?(:clj clojure.test/testing :cljs cljs.test/testing) name (func)))
+
+(defn -report
+  [reports]
+  (#?(:clj clojure.test/report :cljs cljs.test/report) reports))
```

Comment:
If we used a namespace alias for `clojure.test` and `cljs.test` (i.e., the same alias), we wouldn't need the reader conditional here would we?